nilearn/plotting/displays/_figures.py

```diff
@@ -104,3 +130,354 @@ def savefig(self, output_file=None):
+            Every index needs to correspond to one index in roi_map.
+            If :obj:`None`, all regions in roi_map are used.
+
+        labels : :obj:`list` of :obj:`str` or :obj:`None`, default=None
+            A :obj:`list` of labels for the individual regions of interest.
+            Provide :obj:`None` as list entry to skip showing the label of
+            that region. If :obj:`None`, no labels are used.
+
+        lines : :obj:`list` of :obj:`dict` giving the properties of the \
+                contours, or :obj:`None`, default=None
+            For valid keys, see :attr:`plotly.graph_objects.Scatter3d.line`.
+            If length 1, the properties defined in that element will be used
+            to draw all requested contours.
+
+        elevations : :obj:`float`, default=0.1
+            Controls how high above the face each boundary should be placed.
+            0.0 implies directly on boundary, and higher values are farther
+            above the face. This is useful for avoiding overlap of surface
+            and boundary.
+
+        Warnings
+        --------
+            Warns when a vertex is isolated; it will not be included in the
+            roi contour.
+
+        Notes
+        -----
+            Regions are traced by connecting the centroids of non-isolated
+            faces (triangles).
+        """
+        if levels is None:
+            levels = np.unique(roi_map)
+        if labels is None:
+            labels = [f"Region {i}" for i, _ in enumerate(levels)]
+        if lines is None:
+            lines = [None] * len(levels)
+        elif len(lines) == 1 and len(levels) > 1:
+            lines *= len(levels)
+        if not (len(levels) == len(labels)):
+            raise ValueError(
+                "levels and labels need to be either the same length or None."
+            )
+        if not (len(levels) == len(lines)):
+            raise ValueError(
+                "levels and lines need to be either the same length or None."
+            )
+
+        roi = load_surf_data(roi_map)
+
+        traces = []
+        for level, label, line in zip(levels, labels, lines):
+            parc_idx = np.where(roi == level)[0]
+
+            # warn when the edge faces exclude vertices in parcellation
+            # a vertex is isolated when it is a vertex of 6 faces that
+            # each have only on vertex in parcellation
+            verts_per_face = np.isin(self._faces, parc_idx).sum(axis=1)
+            faces_w_one_v = np.flatnonzero(verts_per_face == 1)
+            unique_v, unique_v_counts = np.unique(
+                self._faces[faces_w_one_v], return_counts=True
+            )
+            isolated_v = unique_v[unique_v_counts == 6]
+            if any(isolated_v):
+                warnings.warn(
+                    f"""{label=} contains isolated vertices:
+                    {isolated_v.tolist()}. These will not be included in ROI
+                    boundary line."""
```

Comment:
```suggestion
                    boundary line.""", stacklevel=2
```

having a stacklevel higher than the default will help the users know where it's coming from in their code and not in our code:

https://www.youtube.com/watch?v=CtFdXBEwYfk

Reply:
Oh! Thanks. I wasn't aware of this.

Reply:
No worries. A lot of warning in the Nilearn code base suffer from that problem so it shows you are not the only one not to know this.